sync-team/src/github/mod.rs

```diff
@@ -506,17 +506,21 @@ pub fn construct_branch_protection(
     branch_protection: &rust_team_data::v1::BranchProtection,
 ) -> api::BranchProtection {
     let uses_homu = branch_protection.merge_bots.contains(&MergeBot::Homu);
-    let required_approving_review_count: u8 = if uses_homu {
-        0
+    // When homu manages a branch, we should not require a PR nor approvals
+    // for that branch, because homu pushes to these branches directly.
```

Comment:
but homu is in the allowlist, so why is this change required?
![image](https://github.com/user-attachments/assets/5986a8fb-7e43-4e22-8aa9-146489ff5f46)

Reply:
Conceptually, when a bors manages a branch, PRs (and PR reviews) are not required, because bors pushes to these branches directly. So I think that the branch protections should reflect that (even though GH also allows the bot to push even if PRs are required, since we didn't configure any required CI checks).

Reply:
I see. Sounds good to me. before merging this, is there a way to visualize the branch protection difference in the dry run?

Reply:
Hmm, we currently don't log the branch protection diffs fully. Let me fix that.

Reply:
https://github.com/rust-lang/team/pull/1756